api/python/cellxgene_census/tests/experimental/ml/test_pytorch.py

```diff
@@ -519,6 +519,26 @@ def test_experiment_dataloader__batched(soma_experiment: Experiment, use_eager_f
     assert batch[1].tolist() == [[0, 0], [1, 1], [2, 2]]
 
 
+@pytest.mark.experimental
+# noinspection PyTestParametrized,DuplicatedCode
+@pytest.mark.parametrize(
+    "obs_range,var_range,X_value_gen,use_eager_fetch",
+    [(10, 3, pytorch_x_value_gen, use_eager_fetch) for use_eager_fetch in (True, False)],
+)
+def test_experiment_dataloader__batched_length(soma_experiment: Experiment, use_eager_fetch: bool) -> None:
+    dp = ExperimentDataPipe(
+        soma_experiment,
+        measurement_name="RNA",
+        X_name="raw",
+        obs_column_names=["label"],
+        batch_size=3,
+        shuffle=False,
+        use_eager_fetch=use_eager_fetch,
+    )
+    dl = experiment_dataloader(dp)
+    assert len(dl) == 4
```

Comment:
Could be nice to do this, especially if you were to parameterize for shuffling, other batch sizes etc

```suggestion
    assert len(dl) == len(list(dl))
```